applicationconfig/client/src/test/java/com/azure/applicationconfig/ConfigurationAsyncClientTest.java

```diff
@@ -869,34 +1005,57 @@ private static void assertConfigurationEquals(ConfigurationSetting expected, Res
+     * @return True if the ConfigurationSettings are value equivalent
      */
     private static void assertConfigurationEquals(ConfigurationSetting expected, ConfigurationSetting actual) {
-        if (expected == null) {
-            assertNull(actual);
-            return;
+        if (expected != null && ConfigurationSetting.NO_LABEL.equals(expected.label())) {
+            actual.label(ConfigurationSetting.NO_LABEL);
         }
 
-        assertNotNull(actual);
-        assertEquals(expected.key(), actual.key());
+        assertTrue(configurationsEqual(expected, actual));
+    }
+
+    /**
+     * Helper method to verify that two ConfigurationSettings are datawise equal.
+     * @param expected ConfigurationSetting expected to be returned by the service.
+     * @param actual ConfigurationSetting returned by the service.
+     * @return True if the ConfigurationSettings are the same setting or their member variables are equivalent.
+     */
+    private static boolean configurationsEqual(ConfigurationSetting expected, ConfigurationSetting actual) {
+        if (expected == actual) {
+            return true;
+        }
 
-        // This is because we have the no label which is deciphered in the service as "\0".
-        if (ConfigurationSetting.NO_LABEL.equals(expected.label())) {
-            assertNull(actual.label());
-        } else {
-            assertEquals(expected.label(), actual.label());
+        if (expected == null) {
+            return false;
         }
 
-        assertEquals(expected.value(), actual.value());
-        assertEquals(expected.contentType(), actual.contentType());
+        if (!Objects.equals(expected.key(), actual.key())
+            || !Objects.equals(expected.value(), actual.value())
+            || !Objects.equals(expected.contentType(), actual.contentType())) {
+            return false;
+        }
 
-        if (expected.tags() != null) {
-            assertEquals(expected.tags().size(), actual.tags().size());
+        if (ConfigurationSetting.NO_LABEL.equals(expected.label()) && actual.label() != null) {
+            return false;
+        } else if (!Objects.equals(expected.label(), actual.label())) {
+            return false;
+        }
 
-            expected.tags().forEach((key, value) -> {
-                assertTrue(actual.tags().containsKey(key));
-                assertEquals(value, actual.tags().get(key));
-            });
+        if (expected.tags() != null && expected.tags().size() != 0 && actual.tags() != null) {
+            if (!Objects.equals(expected.tags().size(), actual.tags().size())) {
+                return false;
+            }
+
+            for (Map.Entry<String, String> tag : expected.tags().entrySet()) {
+                if (!actual.tags().containsKey(tag.getKey())
+                    || !Objects.equals(tag.getValue(), actual.tags().get(tag.getKey()))) {
+                    return false;
+                }
```

Comment:
This seems unnecessary. You should be able to compare the two maps equality directly, rather than dealing with it yourself.

Reply:
Removed